File: public/usage-examples/animations/restart_animation_with_sound-1-example.py
from splashkit import *

def main():
    script = load_animation_script('Resources/animations/kermit.txt')
    anim = create_animation_from_script(script, 'WalkFront')

    # Restart animation and let any first-frame sound play
    restart_animation_with_sound(anim, True)
    write_line('Animation restarted (with_sound=true).')

    free_animation(anim)
    free_animation_script(script)

if __name__ == '__main__':
    main()

File: public/usage-examples/animations/restart_animation_with_sound-1-example.cpp
#include "splashkit.h"

int main()
{
    // Load script and create animation
    animation_script script = load_animation_script("Resources/animations/kermit.txt");
    Animation anim = create_animation_from_script(script, "WalkFront");

    // Restart and play sound if the first frame defines one
    restart_animation_with_sound(anim, true);
    write_line("Animation restarted (with_sound=true).");

    free_animation(anim);
    free_animation_script(script);
    return 0;
}
